File: python/phone2action_get_request.py
# We'll use the `requests` module to make our HTTP Request 
import requests

# We'll use the `json` module to format our response's data
import json

# Our Phone2Action API Key goes here
# Super important, we need this to show that we're authorized to make our request
x_api_key = '' # Should not be empty
# See the README for info on how to get an API Key

# First, we define our base URL and our parameters
# Our parameters can be a Dictionary, since the `requests` module converts them
# Our headers can also be a Dictionary, `requests` converts these as well
base_URL = 'https://fmrrixuk32.execute-api.us-east-1.amazonaws.com/hacktj/legislators'
parameters = {
  'level': 'NATIONAL_LOWER',
  'address': '6560 Braddock Rd'
}
headers = {
  'X-API-Key': x_api_key
}

# Now, we're able to fire the configured HTTP GET request
response = requests.get(base_URL, params=parameters, headers=headers)

# Next, we convert the response to a JSON with `json.loads()`
representatives = json.loads(response.text)['officials']

# Finally, we pick out the data we want and print it to the console
# Use Postman to figure out what format your data is in and access it accordingly
# A sample JSON is also available at https://github.com/phone2action/hacktj-examples#legislators-api
for representative in representatives:
    first_name = representative['first_name']
    last_name = representative['last_name']
    print('Your Representative in national Congress is ' + first_name + ' ' + last_name)

File: node/Phone2ActionGETRequest.js
// We'll use Axios to make our HTTP Request, install with NPM or Yarn!
var Axios = require('axios')

// Our Phone2Action API Key goes here
// Super important, we need this to show that we're authorized to make our request
var xAPIKey = '' // Should not be empty, see the README for info on how to get an API Key

Axios.defaults.headers.common['X-API-Key'] = xAPIKey // We apply the API Key to Axios
// We're changing the defaults, so every HTTP Request Axios sends will include the API Key!

// First, we define our base URL and our parameters
// Our parameters can be a JSON, since Axios will convert them
var baseURL = 'https://fmrrixuk32.execute-api.us-east-1.amazonaws.com/hacktj/legislators'
var parameters = {
  'level': 'NATIONAL_LOWER',
  'address': '6560 Braddock Rd'
}

// Now, we're able to fire the configured HTTP GET request
// Axios uses JavaScript Promises, so it'll call `.then()` when done
Axios.get(baseURL, {
  params: parameters
}).then(function(response) {
  var representatives = response.data.officials
  // Finally, we pick out the data we want and print it to the console
  // Use Postman to figure out what format your data is in and access it accordingly
  representatives.forEach(function(representative) {
    var firstName = representative['first_name']
    var lastName = representative['last_name']
    console.log('Your Representative in national Congress is', firstName, lastName)
  })
})